### DISCUSSION TITLE
Is it possible to set the base url of the pages

### DISCUSSION THREAD

**Discussion by spam-n-eggs** [11ty/eleventy#1830] [Q&A] [answered]
For example, I'm using Gitlab Pages and the base url is `https://spam-n-eggs.gitlab.io/the-time-project/`.  The problem is that every link redirects to `https://spam-n-eggs.gitlab.io/<resource-name>/`

Is there a way to feed it a base url to start with.  Eventually this will be a non-issue when I get a domain in front of it, but until then, it's a problem.

**Comment by Ryuno-Ki** [comment, marked as answer]
I think, you are looking for the `pathPrefix` config option: https://www.11ty.dev/docs/config/#deploy-to-a-subdirectory-with-a-path-prefix

**Reply by spam-n-eggs** [reply]
Just tried that and no love.  It doesn't look like it's respecting the pathPrefix setting.

This is what my `.eleventy.js` file looks like.  Maybe I missed something?

```javascript
module.exports = (eleventyConfig) => {
    eleventyConfig.addWatchTarget("_tmp");
    eleventyConfig.addPassthroughCopy( {"_tmp/style.css": "styles/base.css"});
    return {
        pathPrefix: "/the-time-project/"
    }
}
```

**Reply by pdehaan** [reply]
It should definitely work, but there can be a few gotchas.
The subtle-by-very-important bit from the docs is:

> _"It’s used by the `url` filter and inserted at the beginning of all absolute url href links."_

So IIRC, you need to always use the `| url` filter in your links+permalinks if you want it to prefix the specified `pathPrefix` setting.

If you have a public repo I can take a look, but from the looks of https://gitlab.com/users/spam-n-eggs/projects, the project might be private or something.

**Reply by spam-n-eggs** [reply]
Yup.  I just figured out the missing link of using the `| url` filter.  

Thanks!

**Reply by Ryuno-Ki** [reply]
Thanks for the help, @pdehaan!
